packages/core/src/connectors/metaMask.ts

```diff
@@ -0,0 +1,49 @@
+import { Chain } from '../types'
+import { InjectedConnector, InjectedConnectorOptions } from './injected'
+
+export type MetaMaskConnectorOptions = Pick<
+  InjectedConnectorOptions,
+  'shimDisconnect'
+>
+
+export class MetaMaskConnector extends InjectedConnector {
+  readonly id = 'metaMask'
+  readonly ready =
+    typeof window != 'undefined' && !!this.#findProvider(window.ethereum)
+
+  #provider?: Window['ethereum']
+
+  constructor(config?: {
+    chains?: Chain[]
+    options?: MetaMaskConnectorOptions
+  }) {
+    super({
+      ...config,
+      options: {
+        name: 'MetaMask',
+        shimDisconnect: true,
+        ...config?.options,
+      },
+    })
+  }
+
+  async getProvider() {
+    if (typeof window !== 'undefined')
+      this.#provider = this.#findProvider(window.ethereum)
+    return this.#provider
+  }
+
+  #getReady(ethereum?: Ethereum) {
+    const isMetaMask = !!ethereum?.isMetaMask
+    // Brave Wallet tries to make itself look like MetaMask
+    // Agressively try matching against these properties unique to MetaMask
+    if (isMetaMask && !!ethereum._events?.connect && !!ethereum._state)
```

Comment:
This is still needed even though `ethereum.isBraveWallet` is present? (Based on the following lines)

Reply:
Nope, should be good without it. Removed!